lua/rocks/operations/helpers.lua

```diff
@@ -29,11 +29,111 @@ local helpers = {}
+    -- Give preference to class methods/fields
+    if MultiMutRocksTomlWrapper[key] then
+        return MultiMutRocksTomlWrapper[key]
+    end
+    -- Find the key within the config tables
+    local tables = {}
+    for _, tbl in ipairs(self.tables) do
+        if tbl[key] ~= nil then
+            if type(tbl[key]) == "table" then
+                table.insert(tables, tbl[key])
+            else
+                return tbl[key]
+            end
+        end
+    end
+    -- If the value is a table, setup a nested metatable that uses the
+    -- inner tables of the config tables
+    if #tables > 0 then
+        if not self.cache[key] then
+            self.cache[key] = MultiMutRocksTomlWrapper.new(true, tables)
+        end
+        return self.cache[key]
+    end
+    return nil
+end
+MultiMutRocksTomlWrapper.__newindex = function(self, key, value)
+    for _, tbl in ipairs(self.tables) do
+        if tbl[key] ~= nil then
+            tbl[key] = value
+            return
+        end
+    end
+    -- If key not found in any table, add it to the first table
+    self.tables[1][key] = value
+end
+
+--- Write to all rocks toml config files
+function MultiMutRocksTomlWrapper:write()
+    if not self.paths then
+        return
+    end
+    for i, path in ipairs(self.paths) do
+        if self.tables[i] then
+            fs.write_file_await(path, "w", tostring(self.tables[i]))
+        end
+    end
+end
+
+--- Function to create a new wrapper
+---@param nested boolean Whether this is a nested metatable
+---@param tables MutRocksTomlRef[] Rocks toml config metatables
+---@param paths? string[] File paths associated the rocks toml configs
+---@return MultiMutRocksTomlWrapper
+function MultiMutRocksTomlWrapper.new(nested, tables, paths)
+    if #tables < 1 then
+        return {}
+    end
+    if not nested and #tables ~= #paths then
+        return {}
+    end
+    local self = { cache = {}, tables = tables, paths = paths }
+    setmetatable(self, MultiMutRocksTomlWrapper)
+    return self
+end
+
 ---Decode the user rocks from rocks.toml, creating a default config file if it does not exist
----@return MutRocksTomlRef
+---@return MultiMutRocksTomlWrapper
 function helpers.parse_rocks_toml()
-    local config_file = fs.read_or_create(config.config_path, constants.DEFAULT_CONFIG)
-    return require("toml_edit").parse(config_file)
+    local rocks_tomls = {}
+    local paths = {}
+    local visited = {}
+
+    local function parse(file_path, default)
+        -- Don't allow recursive includes
+        if visited[file_path] then
+            return
+        end
+        visited[file_path] = true
+
+        local config_file = fs.read_or_create(file_path, default)
+        local rocks_toml = require("toml_edit").parse(config_file)
+        table.insert(rocks_tomls, rocks_toml)
+        table.insert(paths, file_path)
+
+        -- Follow import paths
+        -- TODO: Remove call to get_imports, this is needed because toml-edit doesn't
```

Comment:
issue: I'm personally not a fan of todo comments, as they tend to get forgotten (the ones we do have here are quite old).
Let's move this to an issue (I guess in toml-edit).

Reply:
Finally had some time to work on the toml-edit change last night: https://github.com/nvim-neorocks/toml-edit.lua/pull/38

Reply:
Merged and released. Shall we update the dependency in this PR?

Reply:
Done